pkg/kubernetes/nulecule.js

```diff
@@ -0,0 +1,152 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+define([
+    "jquery",
+    "base1/cockpit",
+    "docker/docker"
+], function($, cockpit, docker) {
+    "use strict";
+
+    var nulecule = { };
+    var _ = cockpit.gettext;
+
+    function debug(args) {
+        if (window.debugging == "all" || window.debugging == "nulecule")
+            console.debug.apply(console, arguments);
+    }
+
+    /**
+     * NuleculeClient
+     *
+     */
+    function NuleculeClient() {
+        var self = this;
+        self.answers = {};
+        self.install_dir = "";
+
+        self.create_tmp = function create_tmp() {
+            var process = cockpit.spawn(['/bin/sh', '-s']).input("mktemp -p /tmp -d APP_ENTITY.XXXX");
+            return process;
+        };
+
+        self.write_answer_file = function write_answer_file(data) {
+            var dfd = $.Deferred();
+            debug("writing to " + self.install_dir + "/answers.conf");
+            var file = cockpit.file(self.install_dir + "/answers.conf");
+            var req = file.replace(JSON.stringify(data))
+                .done(function(){
+                    dfd.resolve();
+                    file.close();
+                })
+                .fail(function(ex){
+                    console.warn(ex);
+                    dfd.reject(ex);
+                    file.close();
+                });
+
+            var promise = dfd.promise();
+            promise.cancel = function cancel() {
+                req.cancel();
+                file.close();
+            };
+
+            return promise;
+        };
+
+        self.close = function close() {
+            return;
+        };
+
+        self.installrun = function installrun(ptype, tmp_dir, image) {
+            var deferred = $.Deferred();
+            var status = '';
+            var promise;
+            var buffer = '';
+            var process ;
+            
+            if(ptype === "install") {
+                process = cockpit.spawn(["atomicapp", "install", "--destination", tmp_dir, image],{ err: "out", superuser: true });
```

Comment:
Doesn't this need to be changed to use the 'atomic' command directly ... before we can merge this?